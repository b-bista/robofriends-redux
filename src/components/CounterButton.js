import React, { Component } from 'react';

class CounterButton extends Component {

  constructor() {
    super();
    this.state = {
      count: 0
    };
  }
  // shouldComponentUpdate - allows us to override if we must update component
  shouldComponentUpdate(nextProps, nextState) {
    if (this.state.count !== nextState.count) {
      return true;
    }

    return false;
  }

  updateCount = () => {
    // Because updating state is Asynchronous, if your state needs the previous state its best to update like this
    this.setState(state => {
      return {count: this.state.count + 1}
    });
  };

  render() {
    return (
      <button
        id="counter"
        color={this.props.color}
        onClick={this.updateCount}>
        Count: {this.state.count}
      </button>
    );
  }
}

export default CounterButton