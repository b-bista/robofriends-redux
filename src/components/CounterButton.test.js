import { shallow, mount, render} from 'enzyme';
import React from 'react';
import CounterButton from './CounterButton';

it('tests that counterbutton is rendered', () => {
    expect.assertions(1);
    const mockColor = 'red';
    expect(shallow(<CounterButton color={mockColor}/>)).toMatchSnapshot();
})

it('correctly increments the counter', () => {
    const wrapper = shallow(<CounterButton />);
    
    expect.assertions(1);
    //id is important because it targets the element
    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.state().count).toEqual(1);
})