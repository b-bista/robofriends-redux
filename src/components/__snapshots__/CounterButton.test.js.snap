// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests that counterbutton is rendered 1`] = `ShallowWrapper {}`;
